e2e/_suites/kubernetes-autodiscover/kubernetes_test.go

```diff
@@ -0,0 +1,174 @@
+	}, exp)
+}
+
+func (c kubernetesControl) Cleanup(ctx context.Context) error {
+	if c.createdNamespace && c.Namespace != "" {
+		output, err := c.Run(ctx, "delete", "namespace", c.Namespace)
+		if err != nil {
+			return fmt.Errorf("failed to delete namespace %s: %v: %s", c.Namespace, err, output)
+		}
+	}
+	return nil
+}
+
+func (c kubernetesControl) Run(ctx context.Context, runArgs ...string) (output string, err error) {
+	return c.RunWithStdin(ctx, nil, runArgs...)
+}
+
+func (c kubernetesControl) RunWithStdin(ctx context.Context, stdin io.Reader, runArgs ...string) (output string, err error) {
+	shell.CheckInstalledSoftware("kubectl")
+	var args []string
+	if c.config != "" {
+		args = append(args, "--kubeconfig", c.config)
+	}
+	if c.Namespace != "" {
+		args = append(args, "--namespace", c.Namespace)
+	}
+	args = append(args, runArgs...)
+	return shell.ExecuteWithStdin(ctx, ".", stdin, "kubectl", args...)
+}
+
+type kubernetesCluster struct {
+	kindName   string
+	kubeconfig string
+
+	tmpDir string
+}
+
+func (c kubernetesCluster) Kubectl() kubernetesControl {
+	return kubernetesControl{}.WithConfig(c.kubeconfig)
+}
+
+func (c kubernetesCluster) isAvailable(ctx context.Context) error {
+	_, err := c.Kubectl().Run(ctx, "api-versions")
+	return err
+}
+
+func (c *kubernetesCluster) initialize(ctx context.Context) error {
+	err := c.isAvailable(ctx)
+	if err == nil {
+		return nil
+	}
+
+	log.Info("Kubernetes cluster not available, will start one using kind")
+	shell.CheckInstalledSoftware("kind")
+	kindVersion, err := shell.Execute(ctx, ".", "kind", "version")
+	if err != nil {
+		log.WithError(err).Fatal("Failed to get kind version")
+	}
+	log.Infof("Using %s", kindVersion)
+
+	c.tmpDir, err = ioutil.TempDir(os.TempDir(), "test-")
+	if err != nil {
+		log.WithError(err).Fatal("Failed to create temporary directory")
+	}
+
+	name := "kind-" + uuid.New().String()
+	c.kubeconfig = filepath.Join(c.tmpDir, "kubeconfig")
+
+	args := []string{
+		"create", "cluster",
+		"--name", name,
+		"--config", "testdata/kind.yml",
+		"--kubeconfig", c.kubeconfig,
+	}
+	if version, ok := os.LookupEnv("KUBERNETES_VERSION"); ok && version != "" {
+		log.Infof("Installing Kubernetes v%s", version)
+		args = append(args, "--image", "kindest/node:v"+version)
+	}
+	output, err := shell.Execute(ctx, ".", "kind", args...)
+	if err != nil {
+		log.WithError(err).WithField("output", output).Fatal("Failed to create kind cluster")
+		return err
+	}
+	c.kindName = name
+
+	log.Infof("Kubeconfig in %s", c.kubeconfig)
+
+	return nil
+}
+
+func (c *kubernetesCluster) cleanup(ctx context.Context) {
+	if c.kindName != "" {
+		_, err := shell.Execute(ctx, ".", "kind", "delete", "cluster", "--name", c.kindName)
+		if err != nil {
+			log.Warnf("Failed to delete kind cluster %s", c.kindName)
+		}
+		c.kindName = ""
+	}
+	if c.tmpDir != "" {
+		err := os.RemoveAll(c.tmpDir)
+		if err != nil {
+			log.Warnf("Failed to remove temporary directory %s", c.tmpDir)
+		}
+	}
```

Comment:
Wdyt about adding this info to logs? reading Jenkins logs I cannot see if the test suite was finished (OTOH they scenarios are present in the Tests UI)
```suggestion
	}
	log.Infof("kind cluster %s was deleted", c.kindName)
```